JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -210,6 +217,38 @@ internal async Task ListenLoopAsync(CancellationToken ct)
             }
         }
 
+        /// <summary>
+        /// Creates a task that extends the message visibility of the batch on a cadence of half the timeout duration, until all of the messages have been processed.
+        /// </summary>
+        /// <param name="inflightTracker"></param>
+        /// <param name="cancellationToken"></param>
+        private void StartBatchHeartbeat(MessageBatchInflightTracker inflightTracker, CancellationToken cancellationToken)
+        {
+            _ = Task.Run(async () =>
+            {
+                // TODO overall timeout, or max renews? Could renew forever.
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    await Task.Delay(TimeSpan.FromTicks(_queue.VisibilityTimeout.Ticks / 2), cancellationToken);
+
+                    var inflightMessages = inflightTracker.InflightMessages.ToList();
+                    if (inflightMessages.Count == 0) return;
+
+                    await _queue.Client.ChangeMessageVisibilityBatchAsync(new ChangeMessageVisibilityBatchRequest
```

Comment:
I am assuming here that this will take no longer than (visibility / 2) in which case it could get a bit funky.
Perhaps this should be fire and forget so that the cadence is as regular as possible.